surfacers/internal/postgres/postgres.go

```diff
@@ -266,6 +275,14 @@ func (s *Surfacer) init(ctx context.Context) error {
 	go func() {
 		defer s.db.Close()
 
+		metricsBatchMinimumFlushSize := resolveMetricsBatchMinimumFlushSize(s.c.MetricsBatchMinimumFlushSize, DefaultMetricsBatchMinimumFlushSize)
+		metricsBatchFlushIntervalMsec := resolveMetricsBatchFlushInterval(s.c.MetricsBatchFlushIntervalMsec, DefaultMetricsBatchFlushIntervalMsec)
+
+		buffer := make([]*metrics.EventMetrics, 0, metricsBatchMinimumFlushSize)
+		flushInterval := time.Duration(metricsBatchFlushIntervalMsec) * time.Millisecond
+
+		var wait = time.After(flushInterval)
+
```

Comment:
Instead of using a time.After channel and creating a new channel everytime, I'd suggest use a ticker here:

```suggestion
		flushTicker := time.Ticker(flushInterval)
		defer flushTicker.Stop()


```